num_refined.py

```diff
@@ -0,0 +1,43 @@
+import pandas as pd
+from sklearn.preprocessing import MinMaxScaler
+
+
+def load_and_preprocess_data(file_name: str):
+
+    # The function: loads data from an Excel file, removes unnecessary columns,
+    # applies One-Hot Encoding, and scales the 'budget_millions' values.
+
+    # 1. Loads the data from the .xlsx file
+    data = pd.read_excel(file_name)
+
+    # 2. Removes unwanted columns
+    data = data.drop(columns=["row_number", "id"])
+
+    # 3. One-Hot Encoding used on "project_category_codes" and "district_codes"
+    data = pd.get_dummies(data, columns=["project_category_codes", "district_codes"])
+
+    # 4. Converting the boolean values to integer (0 or 1)
+    data = data.astype(int)

```

Comment:
this will convert even budget_millions, is it ok? do not want it to be float if you are scaling it later?
also, scalers should be stored and loaded later to have the same scaling if you load multiple documents

if you want to have simplified code, scale differently, like having arbitrary ranges
0-500k
500k-1M
1M-2M
...
8M+